transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -846,7 +992,16 @@ private void invokeWriteAndFlush(Object msg, ChannelPromise promise) {
     }
 
     private void write(Object msg, boolean flush, ChannelPromise promise) {
-        AbstractChannelHandlerContext next = findContextOutbound();
+        final AbstractChannelHandlerContext next;
+        if (flush) {
+            AbstractChannelHandlerContext ctx = this;
+            do {
+                ctx = ctx.prev;
+            } while ((ctx.executionMask & MASK_WRITE) == 0 && (ctx.executionMask & MASK_FLUSH) == 0);
```

Comment:
Isn't `(ctx.executionMask & MASK_WRITE) == 0 && (ctx.executionMask & MASK_FLUSH) == 0` the same as:
`(ctx.executionMask & (MASK_WRITE|MASK_FLUSH)) == 0`

Reply:
@nitsanw yes! you are right ....